features/harmful-apis/main.js

```diff
@@ -0,0 +1,329 @@
+                .then(data => {
+                    const chData = {
+                        'device-memory': data.headers['device-memory'],
+                        downlink: data.headers.downlink,
+                        dpr: data.headers.dpr,
+                        ect: data.headers.ect,
+                        rtt: data.headers.rtt,
+                        'viewport-width': data.headers['viewport-width']
+                    };
+
+                    if (Object.values(chData).every(i => i === undefined)) {
+                        return false;
+                    }
+
+                    return chData;
+                });
+        }
+    },
+    {
+        id: 'topics-api',
+        run: () => {
+            return ('browsingTopics' in document);
+        }
+    },
+    {
+        id: 'floc-api',
+        run: () => {
+            return ('interestCohort' in document);
+        }
+    },
+    {
+        id: 'raw-sockets-api',
+        run: () => {
+            return ('openUDPSocket' in navigator) || ('openTCPSocket' in navigator);
+        }
+    },
+    {
+        id: 'idle-detection-api',
+        run: () => {
+            return ('IdleDetector' in window);
+        }
+    },
+    {
+        id: 'raw-clipboard-api',
+        run: () => {
+            return ('clipboard' in navigator) && ('platform' in navigator.clipboard);
+        }
+    },
+    {
+        id: 'native-filesystem-api',
+        run: () => {
+            return ('showOpenFilePicker' in window);
+        }
+    },
+    {
+        id: 'background-sync-api',
+        run: () => {
+            return ('ServiceWorkerRegistration' in window) && ('sync' in ServiceWorkerRegistration.prototype);
+        }
+    },
+    {
+        id: 'periodic-sync-api',
+        run: () => {
+            return ('ServiceWorkerRegistration' in window) && ('periodicSync' in ServiceWorkerRegistration.prototype);
+        }
+    },
+    {
+        id: 'push-api',
+        run: () => {
+            return ('ServiceWorkerRegistration' in window) && ('pushManager' in ServiceWorkerRegistration.prototype);
+        }
+    },
+    {
+        id: 'first-party-sets-api',
+        run: () => {
+            if (!window.cookieStore) {
+                return '❌ Can\'t be tested - CookieStore not available';
+            }
+
+            return window.cookieStore.set({ name: 'first-party-sets-test', value: 'value' })
+                .then(() => window.cookieStore.getAll())
+                .then(cookies => {
+                    const cookie = cookies.find(c => c.name === 'first-party-sets-test');
+
+                    return ('sameParty' in cookie);
+                });
+        }
+    }
+];
+
+// object that contains results of all tests
+const results = {
+    page: `${PAGE_NAME}-test`,
+    date: null,
+    results: []
+};
+
+function resultToHTML (data) {
+    return ((data === false) ? '❌ API not available' : '✅ API available') + ' (' + JSON.stringify(data, null, 2) + ')';
```

Comment:
Nit: Since we consider these APIs harmful, shouldn't we flip the icons? Green=good, Red=Bad.